hooks/post-checkout

```diff
@@ -23,16 +24,35 @@ if [[ -n "${BUILDKITE_PLUGIN_TRIVY_SEVERITY:-}" ]] ; then
   export args+=("--severity ${BUILDKITE_PLUGIN_TRIVY_SEVERITY}")
   echo "using non-default severity types"
 fi
+
+if [[ -n "${BUILDKITE_PLUGIN_TRIVY_SECURITY_CHECKS:-}" ]] ; then
+  export fsargs+=("--security-checks ${BUILDKITE_PLUGIN_TRIVY_SECURITY_CHECKS}")
+  echo "using $BUILDKITE_PLUGIN_TRIVY_SECURITY_CHECKS security checks"
+else
+  echo "using default security checks"
+  export fsargs+=("--security-checks vuln,config")
+fi
+
 echo "scanning filesystem"
-docker run -v $PWD:/workdir --rm $image fs ${args[@]} --security-checks vuln,config /workdir
+docker run -v "${PWD}":/workdir --rm "$image" fs "${args[@]}" "${fsargs[@]}" /workdir
+
 
-if [[ -f $PWD/Dockerfile ]]; then
-  docker build -t local/${BUILDKITE_PIPELINE_ID}:${BUILDKITE_COMMIT} .
+if [[ -n "${BUILDKITE_PLUGIN_TRIVY_IMAGE_REF:-}" ]] ; then
+  if [[ "${BUILDKITE_TESTING}" ]]; then
+  	docker pull "${BUILDKITE_PLUGIN_TRIVY_IMAGE_REF}"
+  	echo "using image from parameters"
+  	docker run -v /var/run/docker.sock:/var/run/docker.sock --rm "${image}" image "${args[@]}" "${BUILDKITE_PLUGIN_TRIVY_IMAGE_REF}"
```

Comment:
I'm still new to trivy... what is the purpose of exposing the docker unix socket to the container?

Reply:
I guess I'm continually surprised to see security tools make decisions that undermine other tools' security models (or at least promote practices that, if copied, will be problematic elsewhere).